tensorflow/python/lib/io/file_io.py

```diff
@@ -778,6 +779,24 @@ def list_directory_v2(path):
       for filename in _pywrap_file_io.GetChildren(compat.path_to_bytes(path))
   ]
 
+@tf_export("io.gfile.join")
+def join(path, *paths):
+  """Join one or more path components intelligently.
+
+  This is the same as os.path.join except that it handlers
+  TensorFlow specific filesystems like ram:// and gcs://.
+
+  Args:
+    path: string, path to a directory
+    paths: additional paths to concatenate
+
+  Returns:
+    path: the joined path.
+  """
+  if path.startswith("ram://") or path.startswith("gs://"):
+    return urljoin(path, *paths)
+  return os.path.join(path, *paths)
```

Comment:
I think it's might be more suitable to check `not (path.startswith("://") or path.startswith("file://"))` because we have a lot of cloud filesystem, like `s3`, `hdfs`, etc. WDYT ?

Reply:
I was hoping someone would come along with a better idea :)

It seems like `ram://`, `gc://` or anything else with `://` would not be valid folder or filename is mac, linux or windows (based on [SO](https://stackoverflow.com/a/31976060/6582418)). So maybe we can do:

```python
if `://` in path and not path.startswith("file://")
```

That avoids more complex alternatives like a regex or something.